src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/createExpandedRowHeaders.test.ts

```diff
@@ -0,0 +1,1549 @@
+import Header from '../../components/utils/Header';
+import createExpandedRowHeaders from '../createExpandedRowHeaders';
+
+describe('createExpandedRowHeaders', () => {
+  test('should return an empty array if no row headers are provided', () => {
+    expect(createExpandedRowHeaders([])).toEqual([]);
+  });
+
+  test('should return a single row of row headers if no groups are provided', () => {
+    const rowHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3'),
+    ];
+
+    expect(createExpandedRowHeaders(rowHeaders)).toEqual([
+      [
+        {
+          crossSpan: 1,
+          id: '1',
+          isGroup: false,
+          span: 1,
+          text: '1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '2',
+          isGroup: false,
+          span: 1,
+          text: '2',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '3',
+          isGroup: false,
+          span: 1,
+          text: '3',
+        },
+      ],
+    ]);
+  });
+
+  test('should return multiple rows of row headers if groups are provided', () => {
+    const rowHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3').addChild(new Header('2.1', '2.1')),
+      new Header('4', '4').addChild(new Header('3.1', '3.1')),
+    ];
+
+    expect(createExpandedRowHeaders(rowHeaders)).toEqual([
```

Comment:
We might want to add a type parameter here to get feedback during compilation instead of at runtime:

```ts
.toEqual<ExpandedHeader[][]>(...);
```

Same applies to all the other test cases where we're doing something similar.